### PR TITLE
Feature/dropdown

### PR DESCRIPTION
**Pull request by hafstad**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #184 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Should be decided if "dropdown" is equal to the popover or a select (form-control) with the popover interface:
https://ionicframework.com/docs/api/popover
https://ionicframework.com/docs/api/select

### CODE REVIEW

File: src/kirby/components/dropdown/dropdown.component.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+          scrollContainer.scrollTop = itemTop;
+        } else if (itemBottom > containerVisibleBottom) {
+          scrollContainer.scrollTop = itemBottom - scrollContainer.clientHeight;
+        }
+      }
+    }
+  }
+
+  @HostListener('keydown.tab', ['$event'])
+  private _onTab(event: KeyboardEvent) {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.ts]: It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?

File: src/kirby/components/dropdown/dropdown.component.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    if (this.isOpen) {
+      event.preventDefault();
+      this.close();
+    }
+  }
+
+  @HostListener('keydown.enter')
+  @HostListener('keydown.escape')
+  @HostListener('blur')
+  private _onBlur() {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.ts]: It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?

File: src/kirby/components/dropdown/dropdown.component.ts [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    if (this.isOpen) {
+      this.close();
+    }
+  }
+
+  @HostListener('keydown.space', ['$event'])
+  private _onSpace(event: KeyboardEvent) {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.ts]: It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?

File: src/kirby/components/dropdown/dropdown.component.ts [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    event.preventDefault();
+    event.stopPropagation();
+    if (!this.isOpen) {
+      this.open();
+    }
+  }
+
+  @HostListener('keydown.enter', ['$event'])
+  private _onEnter(event: KeyboardEvent) {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.ts]: It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?

File: src/kirby/components/dropdown/dropdown.component.ts [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+    event.preventDefault();
+    event.stopPropagation();
+    this.toggle();
+  }
+
+  @HostListener('keydown.arrowup', ['$event'])
+  @HostListener('keydown.arrowdown', ['$event'])
+  @HostListener('keydown.arrowleft', ['$event'])
+  @HostListener('keydown.arrowright', ['$event'])
+  private _onArrowKeys(event: KeyboardEvent) {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.ts]: It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?

File: src/kirby/components/dropdown/dropdown.component.ts [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+      }
+    }
+    if (newIndex > -1) {
+      this.selectItem(newIndex);
+    }
+  }
+
+  @HostListener('keydown.home', ['$event'])
+  @HostListener('keydown.end', ['$event'])
+  private _onHomeEndKeys(event: KeyboardEvent) {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.ts]: It's not possible to do AOT-compile `@HostBindings` to private functions - which is a requisite for producing an transpiled version of kirby. Instead - consider making the function public (but keep the underscore prefix to indicate that it should be consider `private`)?

File: src/kirby/components/dropdown/dropdown.component.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  kirby-button
+  [attentionLevel]="isOpen ? attentionLevelOpen : attentionLevel"
+  tabindex="-1"
+  (click)="onToggle()"
+  (mousedown)="onButtonMouseEvent($event)"
+>
+  {{ selectedText }}
+  <kirby-icon [name]="isOpen ? 'arrow-up' : 'arrow-down'"></kirby-icon>
+</button>
+<kirby-card>
```
Comment (MortenGregersen) [inline src/kirby/components/dropdown/dropdown.component.html]: The list of options should have a `listbox` role and every option should have an `option` role (see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role)

File: src/kirby/components/dropdown/dropdown.component.scss [hunk 8]
```diff
@@ -0,0 +1,50 @@
+@import '../../scss/utils';
+
+$item-height: 56px;
+$dropdown-max-height: 5 * $item-height;
```
[the remark this answers is not in the document]
Comment (jakobe) [reply src/kirby/components/dropdown/dropdown.component.scss]: Yes, but we don't support landscape at the moment :)

File: src/kirby/components/dropdown/dropdown.component.ts (hunk 2 above)
Comment (jakobe) [reply src/kirby/components/dropdown/dropdown.component.ts]: Ok, good point 👍

File: src/kirby/components/dropdown/dropdown.component.html (hunk 7 above)
Comment (jakobe) [reply src/kirby/components/dropdown/dropdown.component.html]: The list already has the `listbox` role (see: https://github.com/kirbydesign/designsystem/pull/899/files#diff-a963934da666cc58e9394c82c15b0d91R43)
Good point regarding the `optionn` role, I'll add that 👍

Comment (jakobe) [reply src/kirby/components/dropdown/dropdown.component.html]: Added 👍

File: src/kirby/components/dropdown/dropdown.component.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (jakobe) [reply src/kirby/components/dropdown/dropdown.component.scss]: image
@askarby ☝️ 😄

[the remark this answers is not in the document]
Comment (askarby) [reply src/kirby/components/dropdown/dropdown.component.scss]: giphub

File: src/app/showcase/dropdown-showcase/dropdown-showcase.component.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+import { ShowcaseProperty } from '~/app/shared/showcase-properties/showcase-property';
+
+@Component({
+  selector: 'kirby-dropdown-showcase',
+  templateUrl: './dropdown-showcase.component.html',
+  styleUrls: ['./dropdown-showcase.component.scss'],
+})
+export class DropdownShowcaseComponent {
+  properties: ShowcaseProperty[] = [];
+  constructor() {}
```
[the remark this answers is not in the document]
Comment (jakobe) [reply src/app/showcase/dropdown-showcase/dropdown-showcase.component.ts]: Done

File: src/app/showcase/dropdown-showcase/dropdown-showcase.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
[the remark this answers is not in the document]
Comment (jakobe) [reply src/app/showcase/dropdown-showcase/dropdown-showcase.component.spec.ts]: Done

File: src/app/examples/dropdown-example/dropdown-example.component.spec.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
[the remark this answers is not in the document]
Comment (jakobe) [reply src/app/examples/dropdown-example/dropdown-example.component.spec.ts]: Done

File: src/kirby/components/dropdown/dropdown.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,826 @@
...
+    const button = spectator.query<HTMLButtonElement>('button');
+    expect(button.disabled).toBeFalsy();
+  });
+
+  it('should not render disabled attribute on button', () => {
+    const button = spectator.query<HTMLButtonElement>('button');
+    expect(button.attributes['disabled']).toBeUndefined();
+  });
+
+  it('should recieve focus', () => {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.spec.ts]:
```suggestion
  it('should receive focus', () => {
```

File: src/kirby/components/dropdown/dropdown.component.spec.ts [hunk 13]
```diff
@@ -0,0 +1,826 @@
...
+    spectator.element.focus();
+    spectator.detectChanges();
+    expect(spectator.element).toBeFocused();
+  });
+
+  it('should recieve focus on button click', () => {
```
Comment (askarby) [inline src/kirby/components/dropdown/dropdown.component.spec.ts]:
```suggestion
  it('should receive focus on button click', () => {
```

**Pull request merged**